meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,757 +10,1031 @@ import {
+	const updateExclusivityGroupId = React.useCallback(
+		(newGroupId: string) => {
+			exclusivityOverrideHelper.changeItemId(exclusivityGroup.id, newGroupId)
+			toggleExpanded(newGroupId, true)
+		},
+		[exclusivityOverrideHelper, toggleExpanded, exclusivityGroup.id]
+	)
+
+	return (
+		<React.Fragment>
+			<tr
+				className={ClassNames({
+					hl: isExpanded,
+				})}
+			>
+				<th className="settings-studio-device__name c3">{exclusivityGroup.id}</th>
+				<td className="settings-studio-device__id c5">{exclusivityGroup.computed?.name}</td>
+				<td className="settings-studio-device__id c3">
+					{
+						routeSetsFromOverrides.filter(
+							(routeSet) => routeSet.computed?.exclusivityGroup === exclusivityGroup.computed?.name
+						).length
+					}
+				</td>
+
+				<td className="settings-studio-device__actions table-item-actions c3">
+					<button className="action-btn" onClick={() => toggleExpanded(exclusivityGroup.id)}>
+						<FontAwesomeIcon icon={faPencilAlt} />
+					</button>
+					<button className="action-btn" onClick={confirmRemoveEGroup}>
+						<FontAwesomeIcon icon={faTrash} />
+					</button>
+				</td>
+			</tr>
+			{isExpanded && (
+				<tr className="expando-details hl">
+					<td colSpan={6}>
+						<div className="properties-grid">
+							<label className="field">
+								<LabelActual label={t('Exclusivity Group ID')} />
+								<TextInputControl
+									modifiedClassName="bghl"
+									classNames="input text-input input-l"
+									value={exclusivityGroup.id}
+									handleUpdate={updateExclusivityGroupId}
+									disabled={!!exclusivityGroup.defaults}
+								/>
+							</label>
+							<LabelAndOverrides
+								label={t('Exclusivity Group Name')}
+								item={exclusivityGroup}
+								itemKey={'name'}
+								opPrefix={exclusivityGroup.id}
+								overrideHelper={exclusivityOverrideHelper}
+							>
+								{(value, handleUpdate) => (
+									<TextInputControl
+										modifiedClassName="bghl"
+										classNames="input text-input input-l"
+										value={value}
+										handleUpdate={handleUpdate}
+									/>
+								)}
+							</LabelAndOverrides>
+						</div>
+						<div className="mod alright">
+							<button className="btn btn-primary" onClick={() => toggleExpanded(exclusivityGroup.id)}>
+								<FontAwesomeIcon icon={faCheck} />
+							</button>
+						</div>
+					</td>
+				</tr>
+			)}
+		</React.Fragment>
+	)
+}
+
+interface IRenderExclusivityDeletedGroupProps {
+	exclusivityGroup: WrappedOverridableItemDeleted<StudioRouteSetExclusivityGroup>
+	exlusivityOverrideHelper: OverrideOpHelper
+}
+
+function RenderExclusivityDeletedGroup({
+	exclusivityGroup,
+	exlusivityOverrideHelper: overrideHelper,
+}: Readonly<IRenderExclusivityDeletedGroupProps>): React.JSX.Element {
+	const doUndelete = (groupId: string) => {
+		overrideHelper.resetItem(groupId)
+	}
+
+	const doUndeleteItem = React.useCallback(() => doUndelete(exclusivityGroup.id), [doUndelete, exclusivityGroup.id])
```

Comment:
This `React.useCallback` is doing nothing, as each time it is called `doUndelete` is guaranteed to be a new function.

If you inline the function, and set `overrideHelper` as the dependency, then it will memoize correctly:

```suggestion
	const doUndeleteItem = React.useCallback(() => overrideHelper.resetItem(exclusivityGroup.id), [overrideHelper, exclusivityGroup.id])
```